controllers/extendeddaemonset/controller.go

```diff
@@ -568,11 +569,20 @@ func (r *Reconciler) cleanupReplicaSet(logger logr.Logger, rsList *datadoghqv1al
 }
 
 // shouldDeleteERS returns true if the ers is nil or has no pods attached based on its status.
-func shouldDeleteERS(ers *datadoghqv1alpha1.ExtendedDaemonSetReplicaSet) bool {
+func shouldDeleteERS(now time.Time, ers *datadoghqv1alpha1.ExtendedDaemonSetReplicaSet) bool {
 	if ers == nil {
 		return true
 	}
 
+	// If canary deploy has failed, delay ERS deletion for 5m so that failed metric reports
+	if ersconditions.IsConditionTrue(&ers.Status, datadoghqv1alpha1.ConditionTypeCanaryFailed) {
+		failedIdx := ersconditions.GetIndexForConditionType(&ers.Status, datadoghqv1alpha1.ConditionTypeCanaryFailed)
+		t := ers.Status.Conditions[failedIdx].LastTransitionTime.Add(time.Minute * 5)
+		if now.Before(t) {
+			return false
+		}
+	}
```

Comment:
I guess there are no side effects to this delay? 🤔  In any case, 5m seem a bit long if we consider the agent check interval which is 15s. 1 or 2 minutes should be enough to collect multiple data points.

Reply:
no side effects! the active ERS reconciles as expected. i guess i went with 5m to compensate for the fact it is hard/impossible to see the points on the dashboard currently. but we can do 2m and see how that feels. thanks for the review!